extensions/aerosol_optics/mo_aerosol_optics.F90

```diff
@@ -0,0 +1,681 @@
+      !   See also the increment routines in mo_optical_props_kernels
+      !
+      write(61,*) 'Aerosol Optics: computing final optics...'
+      select type(optical_props)
+      type is (ty_optical_props_1scl)
+        !$acc parallel loop gang vector default(none) collapse(3) &
+        !$acc               copyin(optical_props) copyout(optical_props%tau)
+        !$omp target teams distribute parallel do simd collapse(3) &
+        !$omp map(from:optical_props%tau)
+
+        do ibnd = 1, nbnd
+          do ilay = 1, nlay
+            do icol = 1, ncol
+              ! Absorption optical depth  = (1-ssa) * tau = tau - taussa
+              optical_props%tau(icol,ilay,ibnd) = (atau(icol,ilay,ibnd) - ataussa(icol,ilay,ibnd))
+!              optical_props%tau(icol,ilay,ibnd) = (ltau(icol,ilay,ibnd) - ltaussa(icol,ilay,ibnd)) + &
+!                                                  (itau(icol,ilay,ibnd) - itaussa(icol,ilay,ibnd))
+            end do
+          end do
+        end do
+      type is (ty_optical_props_2str)
+        !$acc parallel loop gang vector default(none) collapse(3) &
+        !$acc               copyin(optical_props) copyout(optical_props%tau, optical_props%ssa, optical_props%g)
+        !$omp target teams distribute parallel do simd collapse(3) &
+        !$omp map(from:optical_props%tau, optical_props%ssa, optical_props%g)
+        do ibnd = 1, nbnd
+          do ilay = 1, nlay
+            do icol = 1, ncol
+              tau    = atau   (icol,ilay,ibnd)
+              taussa = ataussa(icol,ilay,ibnd)
+              optical_props%g  (icol,ilay,ibnd) = (ataussag(icol,ilay,ibnd)) / &
+                                                        max(epsilon(tau), taussa)
+              optical_props%ssa(icol,ilay,ibnd) = taussa/max(epsilon(tau), tau)
+              optical_props%tau(icol,ilay,ibnd) = tau
+!              tau    = ltau   (icol,ilay,ibnd) + itau   (icol,ilay,ibnd)
+!              taussa = ltaussa(icol,ilay,ibnd) + itaussa(icol,ilay,ibnd)
+!              optical_props%g  (icol,ilay,ibnd) = (ltaussag(icol,ilay,ibnd) + itaussag(icol,ilay,ibnd)) / &
+!                                                        max(epsilon(tau), taussa)
+!              optical_props%ssa(icol,ilay,ibnd) = taussa/max(epsilon(tau), tau)
+!              optical_props%tau(icol,ilay,ibnd) = tau
+            end do
+          end do
+        end do
+      type is (ty_optical_props_nstr)
+        error_msg = "aerosol optics: n-stream calculations not yet supported"
+      end select
+    end if 
+    !$acc end data
+    !$omp end target data
+  end function aerosol_optics
+  !--------------------------------------------------------------------------------------------------------------------
+  !
+  ! Inquiry functions
+  !
+  !--------------------------------------------------------------------------------------------------------------------
+  function get_aerosol_size_bin(nbin, aero_bin_lo, aero_bin_hi, aero_size) result(ibin)
```

Comment:
Can you please add comments to say what this code is intended to do?

Reply:
Function purpose comment added.